code/__HELPERS/turfs.dm

```diff
@@ -389,6 +389,25 @@ Turf and target are separate in case you want to teleport some distance from a t
 
 	return get_safe_random_station_turf(final_area_list)
 
+///Returns a list of all safe station turfs, useful for if you would need to do a bunch of calls of get_safe_station_turf()
+///care_about_dense_objects will make us iterate through EVERY OBJ IN EVERY VALID TURF, only set to true if absolutely needed
+/proc/get_safe_station_turfs(list/areas_to_pick_from = GLOB.the_station_areas, care_about_dense_objects = FALSE, allow_outdoor_areas = FALSE) as /list
+	RETURN_TYPE(/list)
+	. = list()
+	for(var/area/turf_area as anything in areas_to_pick_from)
+		var/list/turf_list = get_area_turfs(turf_area)
+		for(var/turf/checked_turf in turf_list)
+			if(!checked_turf.density && (turf_area::area_flags & VALID_TERRITORY) && !isgroundlessturf(checked_turf) && (allow_outdoor_areas || !turf_area.outdoors))
```

Comment:
shouldn't you check `turf_area::area_flags & VALID_TERRITORY` *before* getting the area's turfs (and just continue if so)

Reply:
oh yeah good point, copy paste error moment